
####---- EXPERIMENTHUB METADATA TABLE ----####

meta <- list(
  data.frame(
    Title = "specht2019v2",
    Description = paste0("Specht et al. 2019: macrophages vs monocytes\n",
                         "Single cell proteomics data acquired by the Slavov",
                         "Lab. This is the version 2 of the data released in ", 
                         "December 2019. It contains quantitative information ", 
                         "of monocytes (U-937) and macrophages at PSM, peptide ",
                         "and protein level."),
    BiocVersion = "3.11",
    Genome = NA_character_,
    SourceType = "CSV",
    SourceUrl = "https://scope2.slavovlab.net/docs/data",
    SourceVersion = NA_character_,
    Species = "Homo sapiens",
    TaxonomyId = 9606,
    Coordinate_1_based = TRUE,
    DataProvider = "SlavovLab website",
    Maintainer = "Christophe Vanderaa <christophe.vanderaa@uclouvain.be>",
    RDataClass = "QFeatures",
    DispatchClass = "Rda",
    RDataPath = "scpdata/specht2019v2.Rda",
    PublicationDate = as.Date("2019/12/5"),
    NumberAssays = 179,
    Notes=NA_character_
  ),
  data.frame(
    Title = "specht2019v3",
    Description = paste0("Specht et al. 2019: macrophages vs monocytes\n",
                         "Single cell proteomics data acquired by the Slavov",
                         "Lab. This is the version 3 of the data released in ", 
                         "August 2019. It contains quantitative information ", 
                         "of monocytes (U-937) and macrophages at PSM, peptide ",
                         "and protein level."),
    BiocVersion = "3.11",
    Genome = NA_character_,
    SourceType = "CSV",
    SourceUrl = "https://scope2.slavovlab.net/docs/data",
    SourceVersion = NA_character_,
    Species = "Homo sapiens",
    TaxonomyId = 9606,
    Coordinate_1_based = TRUE,
    DataProvider = "SlavovLab website",
    Maintainer = "Christophe Vanderaa <christophe.vanderaa@uclouvain.be>",
    RDataClass = "QFeatures",
    DispatchClass = "Rda",
    RDataPath = "scpdata/specht2019v2.Rda",
    PublicationDate = as.Date("2019/12/5"),
    NumberAssays = 179,
    Notes=NA_character_
  ),
  data.frame(
    Title = "dou2019_lysates",
    Description = paste0("Dou et al. 2019: HeLa lysates\n",
                         "Single-cell proteomics using nanoPOTS combined with ",
                         "TMT isobaric labeling. It contains quantitative ",
                         "information at PSM and protein level. The samples ", 
                         "are commercial Hela digest diluted to single cell ", 
                         "amounts (0.2ng). The boosting wells contain the same ",
                         "digest but at higher dose (10 ng)."),
    BiocVersion = "3.11",
    Genome = NA_character_,
    SourceType = "XLS/XLSX",
    SourceUrl = "ftp://massive.ucsd.edu/MSV000084110/",
    SourceVersion = NA_character_,
    Species = "Homo sapiens",
    TaxonomyId = 9606,
    Coordinate_1_based = TRUE,
    DataProvider = "MassIVE",
    Maintainer = "Christophe Vanderaa <christophe.vanderaa@uclouvain.be>",
    RDataClass = "QFeatures",
    DispatchClass = "Rda",
    RDataPath = "scpdata/dou2019_lysates.Rda",
    PublicationDate = as.Date("2019/10/15"),
    NumberAssays = 3,
    Notes=NA_character_
  ),
  data.frame(
    Title = "dou2019_mouse",
    Description = paste0("Dou et al. 2019: single cells from cultured murine ",
                         "cell lines.\n",
                         "Single-cell proteomics using nanoPOTS combined with ",
                         "TMT isobaric labeling. It contains quantitative ",
                         "information at PSM and protein level. The cell ",
                         "types are either 'Raw' (macrophage cells), 'C10' ",
                         "(epihelial cells), or 'SVEC' (endothelial cells). ",
                         "Out of the 132 wells, 72 contain single cells, ",
                         "corresponding to 24 C10 cells, 24 RAW cells, and 24 ",
                         "SVEC. The other wells are either boosting channels ",
                         "(12), empty channels (36) or reference channels ",
                         "(12). Boosting and reference channels are balanced ",
                         "(1:1:1) mixes of C10, SVEC, and RAW samples at 5 ng ",
                         "and 0.2 ng, respectively. The different cell types ",
                         "where evenly distributed across 4 nanoPOTS chips. ",
                         "Samples were 11-plexed with TMT labeling."),
    BiocVersion = "3.11",
    Genome = NA_character_,
    SourceType = "XLS/XLSX",
    SourceUrl = "ftp://massive.ucsd.edu/MSV000084110/",
    SourceVersion = NA_character_,
    Species = "Mus musculus",
    TaxonomyId = 10090,
    Coordinate_1_based = TRUE,
    DataProvider = "MassIVE",
    Maintainer = "Christophe Vanderaa <christophe.vanderaa@uclouvain.be>",
    RDataClass = "QFeatures",
    DispatchClass = "Rda",
    RDataPath = "scpdata/dou2019_mouse.Rda",
    PublicationDate = as.Date("2019/10/15"),
    NumberAssays = 13,
    Notes=NA_character_
  ),
  data.frame(
    Title = "dou2019_boosting",
    Description = paste0("Dou et al. 2019: testing boosting ratios.\n",
                         "Single-cell proteomics using nanoPOTS combined with ",
                         "TMT isobaric labeling. It contains quantitative ",
                         "information at PSM and protein level. The cell types ",
                         "are either 'Raw' (macrophage cells), 'C10' ",
                         "(epihelial cells), or 'SVEC' (endothelial cells). ",
                         "Each cell is replicated 2 or 3x. Each cell type was ",
                         "run using 3 levels of boosting: 0 ng (no boosting), ",
                         "5 ng or 50 ng. When boosting was applied, 1 ",
                         "reference well and 1 boosting well were added, ",
                         "otherwise 1 empty well was added. Each boosting ",
                         "setting (0ng, 5ng, 50ng) was run in duplicate."),
    BiocVersion = "3.11",
    Genome = NA_character_,
    SourceType = "XLS/XLSX",
    SourceUrl = "ftp://massive.ucsd.edu/MSV000084110/",
    SourceVersion = NA_character_,
    Species = "Mus musculus",
    TaxonomyId = 10090,
    Coordinate_1_based = TRUE,
    DataProvider = "MassIVE",
    Maintainer = "Christophe Vanderaa <christophe.vanderaa@uclouvain.be>",
    RDataClass = "QFeatures",
    DispatchClass = "Rda",
    RDataPath = "scpdata/dou2019_boosting.Rda",
    PublicationDate = as.Date("2019/10/15"),
    NumberAssays = 7,
    Notes=NA_character_
  ),
  data.frame(
    Title = "zhu2018MCP",
    Description = paste0("Zhu et al. 2018 (Mol. Cel. Prot.): rat brain laser ", 
                         "dissections\n", 
                         "Near single-cell proteomics data of laser cuptured ",
                         "micro-dissection samples. The samples are 24 brain ",
                         "section from rat pups (day 17). The slices are 12 µm ",
                         "thick squares of either 50, 100, or 200 µm width. 5 ",
                         "samples were dissected from the corpus callum (`CC`), ",
                         "4 samples were dissected from the corpus collosum ",
                         "(`CP`), 13 samples were extracted from the cerebral ",
                         "cortex (`CTX`), and 2 samples are labeled as (`Mix``)."),
    BiocVersion = "3.11",
    Genome = NA_character_,
    SourceType = "TXT",
    SourceUrl = "ftp://ftp.pride.ebi.ac.uk/pride/data/archive/2018/07/PXD008844",
    SourceVersion = NA_character_,
    Species = "Rattus norvegicus",
    TaxonomyId = 10116,
    Coordinate_1_based = TRUE,
    DataProvider = "PRIDE",
    Maintainer = "Christophe Vanderaa <christophe.vanderaa@uclouvain.be>",
    RDataClass = "QFeatures",
    DispatchClass = "Rda",
    RDataPath = "scpdata/zhu2018MCP.Rda",
    PublicationDate = as.Date("2018/09/01"),
    NumberAssays = 1,
    Notes=NA_character_
  ),
  data.frame(
    Title = "zhu2018NC_hela",
    Description = paste0("Zhu et al. 2018 (Nat. Comm.): HeLa titration\n",
                         "Near single-cell proteomics data of HeLa samples ",
                         "containing different number of cells. There are ", 
                         "three groups of cell concentrations: low (10-14 ", 
                         "cells), medium (35-45 cells) and high (137-141 ", 
                         "cells). The data also contains measures for blanks, ", 
                         "HeLa lysates (50 cell equivalent) and 2 cancer cell ", 
                         "line lysates (MCF7 and THP1, 50 cell equivalent)."),
    BiocVersion = "3.11",
    Genome = NA_character_,
    SourceType = "TXT",
    SourceUrl = "ftp://ftp.pride.ebi.ac.uk/pride/data/archive/2018/01/PXD00684",
    SourceVersion = NA_character_,
    Species = "Homo sapiens",
    TaxonomyId = 9606,
    Coordinate_1_based = TRUE,
    DataProvider = "PRIDE",
    Maintainer = "Christophe Vanderaa <christophe.vanderaa@uclouvain.be>",
    RDataClass = "QFeatures",
    DispatchClass = "Rda",
    RDataPath = "scpdata/zhu2018NC_hela.Rda",
    PublicationDate = as.Date("2018/02/28"),
    NumberAssays = 1,
    Notes=NA_character_
  ),
  data.frame(
    Title = "zhu2018NC_lysates",
    Description = paste0("Zhu et al. 2018 (Nat. Comm.): HeLa lysates\n", 
                         "Near single-cell proteomics data of HeLa lysates at ", 
                         "different concentrations (10, 40 and 140 cell ", 
                         "equivalent). Each concentration is acquired in ",
                         "triplicate."),
    BiocVersion = "3.11",
    Genome = NA_character_,
    SourceType = "TXT",
    SourceUrl = "ftp://ftp.pride.ebi.ac.uk/pride/data/archive/2018/01/PXD00684",
    SourceVersion = NA_character_,
    Species = "Homo sapiens",
    TaxonomyId = 9606,
    Coordinate_1_based = TRUE,
    DataProvider = "PRIDE",
    Maintainer = "Christophe Vanderaa <christophe.vanderaa@uclouvain.be>",
    RDataClass = "QFeatures",
    DispatchClass = "Rda",
    RDataPath = "scpdata/zhu2018NC_lysates.Rda",
    PublicationDate = as.Date("2018/02/28"),
    NumberAssays = 1,
    Notes=NA_character_
  ),
  data.frame(
    Title = "zhu2018NC_islets",
    Description = paste0("Zhu et al. 2018 (Nat. Comm.): human pancreatic ", 
                         "islets\n", 
                         "Near single-cell proteomics data human pancreas ",
                         "samples. The samples are were laser dissected from ", 
                         "pancreatic tissue slices from pancreata obtained ",
                         "from organ donors through the JDRFNetwork for ", 
                         "Pancreatic Organ Donors with Diabetes (nPOD) ",
                         "program. The sample come either from control ", 
                         "patients (n=9) or from type 1 diabetes (T1D) ",
                         "patients (n=9). "),
    BiocVersion = "3.11",
    Genome = NA_character_,
    SourceType = "TXT",
    SourceUrl = "ftp://ftp.pride.ebi.ac.uk/pride/data/archive/2018/01/PXD00684",
    SourceVersion = NA_character_,
    Species = "Homo sapiens",
    TaxonomyId = 9606,
    Coordinate_1_based = TRUE,
    DataProvider = "PRIDE",
    Maintainer = "Christophe Vanderaa <christophe.vanderaa@uclouvain.be>",
    RDataClass = "QFeatures",
    DispatchClass = "Rda",
    RDataPath = "scpdata/zhu2018NC_islets.Rda",
    PublicationDate = as.Date("2018/02/28"),
    NumberAssays = 1,
    Notes=NA_character_
  ),
  data.frame(
    Title = "cong2020AC",
    Description = paste0("Cong et al. 2020 (Ana. Chem.): HeLa single ", 
                         "cells\n",
                         "Single-cell proteomics using the nanoPOTS ", 
                         "sample processing device in combination with ",
                         "ultranarrow-bore (20μm i.d.) packed-column LC ",
                         "separations and the Orbitrap Eclipse Tribrid MS. ",
                         "The dataset contains label-free quantitative ",
                         "information at PSM, peptide and protein level. ",
                         "The samples are single Hela cells. Bulk samples ",
                         "(100 and 20 cells) were also included in the ",
                         "experiment to increase the idendtification rate ",
                         "thanks to between-run matching (cf MaxQuant)."),
    BiocVersion = "3.11",
    Genome = NA_character_,
    SourceType = "TXT",
    SourceUrl = "ftp://ftp.pride.ebi.ac.uk/pride/data/archive/2020/02/PXD016921",
    SourceVersion = NA_character_,
    Species = "Homo sapiens",
    TaxonomyId = 9606,
    Coordinate_1_based = TRUE,
    DataProvider = "PRIDE",
    Maintainer = "Christophe Vanderaa <christophe.vanderaa@uclouvain.be>",
    RDataClass = "QFeatures",
    DispatchClass = "Rda",
    RDataPath = "scpdata/cong2020AC.Rda",
    PublicationDate = as.Date("2020/01/02"),
    NumberAssays = 9,
    Notes=NA_character_
  ),
  data.frame(
    Title = "zhu2019EL",
    Description = paste0("Zhu et al. 2019 (eLife): chicken utricle ", 
                         "cells\n", 
                         "Near single-cell proteomics data from ", 
                         "chicken utricle acquired to study the ", 
                         "hair-cell development. The cells are ", 
                         "isolated from peeled utrical epithelium and ", 
                         "separated into hair cells (FM1-43 high) and ", 
                         "supporting cells (FM1-43 low). The sample ", 
                         "contain either 1 cell (n = 28), 3 cells ",
                         "(n = 7), 5 cells (n = 8) or 20 cells (n = 14)."),
    BiocVersion = "3.11",
    Genome = NA_character_,
    SourceType = "TXT",
    SourceUrl = "ftp://ftp.pride.ebi.ac.uk/pride/data/archive/2019/11/PXD014256",
    SourceVersion = NA_character_,
    Species = "Gallus gallus",
    TaxonomyId = 9031,
    Coordinate_1_based = TRUE,
    DataProvider = "PRIDE",
    Maintainer = "Christophe Vanderaa <christophe.vanderaa@uclouvain.be>",
    RDataClass = "QFeatures",
    DispatchClass = "Rda",
    RDataPath = "scpdata/zhu2019EL.Rda",
    PublicationDate = as.Date("2020/11/04"),
    NumberAssays = 62,
    Notes=NA_character_
  )
)
meta <- do.call(rbind, meta)
  
write.csv(meta, "inst/extdata/metadata.csv")

